sound/soc/sof/ipc.c

```diff
@@ -269,12 +269,12 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 	/* add message to transmit list */
 	list_add_tail(&msg->list, &ipc->tx_list);
 
+	spin_unlock_irq(&sdev->ipc_lock);
+
```

Comment:
are you sure about this move? that seems quite radical to me.

Reply:
@plbossart its actually not that radical. There are exactly 2 calls to snd_sof_ipc_msgs_tx() in the SOF driver, one of which was in the ipc_lock context and other wasnt. Im just making the change to make it consistent

Reply:
@ranj063 I agree with you. snd_sof_ipc_msgs_tx could be called by multi-threads, and snd_sof_ipc_msgs_tx makes sure there is only one ipc msg could be sent to FW. And please remove " if (snd_sof_dsp_is_ipc_ready(sdev)) " which is checked in snd_sof_ipc_msgs_tx.

Reply:
@RanderWang I think we should keep the check here. Otherwise we'd be scheduling the task unnecessarily.

Reply:
There is no great effect about keeping or removing it. I just want to check snd_sof_dsp_is_ipc_ready at only one point